### DISCUSSION TITLE
Mocha with both WDIO and Frisby

### DISCUSSION THREAD

**Discussion by bajipogadadanda** [webdriverio/webdriverio#13618] [General]
Hi,
We have been using WDIO with Mocha for UI tests automation, now we need to develop webservice tests also along with UI in the same Repo/Framework. So, I decided to use Frisby with Mocha/WDIO. It is working, however, it has been observed that, the Webbrowser instance is being triggered for the web service test case also. Since we are using the WDIO config file, the web browser instance may be triggered. We don't need the browser instance for web service tests, how can we eliminate it? anything can be done for WDIO config file, or all altogether do we need to use a new config file for Frisby tests alone?
Can you provide any references or clues to resolve this issue?

**Comment by christian-bromann** [comment]
> Can you provide any references or clues to resolve this issue?

WebdriverIO always starts a browser to either run functional tests or component tests. If you run pure unit tests we recommend to use Mocha directly or other tools like Vitest or Jest.

**Comment by troyyer** [comment]
a hacky solution would be to have a separate wdio config file for your webservices tests and for these tests have the driver run in headless mode.
